vocabularies/Common.json

```diff
@@ -1288,6 +1288,106 @@
+    },
+    "Template_RecommendationsFunction": [
+      {
+        "$Kind": "Function",
+        "$IsBound": true,
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for functions that compute AI-based recommendations for the bound entity and its related entities",
+        "@Core.LongDescription": "If the function is invoked repeatedly in a stateful session, it SHOULD omit\n          recommendations whose predictors have not changed since the previous invocation.\n\n          The template function itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "Entity",
+            "$Type": "Edm.EntityType",
+            "@Core.Description": "Entity relative to which the URLs in `Targets` are interpreted"
+          },
+          {
+            "$Name": "Targets",
+            "$Collection": true,
+            "@Core.IsURL": true,
+            "@Core.Description": "URLs (relative to the value of the binding parameter) that address entities, properties or action/function parameters for which recommendations shall be retrieved"
+          }
+        ],
+        "$ReturnType": {
+          "@Core.IsURL": true,
+          "@Core.Description": "URL for retrieving the recommendations (which are computed asynchronously)",
+          "@Core.Example": {
+            "@odata.type": "https://oasis-tcs.github.io/odata-vocabularies/vocabularies/Org.OData.Core.V1.xml#Core.PrimitiveExampleValue",
+            "Description": "Relative URL for invoking a function import of the [retrieval function](#Template_RecommendationsRetrievalFunction)",
+            "Value": "../RetrieveRecommendations?ID=af43c2"
+          }
+        }
+      }
+    ],
+    "Template_RecommendationsRetrievalFunction": [
+      {
+        "$Kind": "Function",
+        "@Common.Experimental": true,
+        "@Core.Description": "Template for functions that retrieve AI-based recommendations for the bound entity and its related entities",
+        "@Core.LongDescription": "The template function itself cannot be invoked.",
+        "$Parameter": [
+          {
+            "$Name": "ID",
+            "@Core.Description": "ID that occurs in the URL returned by a prior invocation of the [recommendations function](#Template_RecommendationsFunction)"
+          }
+        ],
+        "$ReturnType": {
+          "$Collection": true,
+          "$Type": "Common.PropertyRecommendationType",
+          "@Core.Description": "A collection of recommendations per targeted property"
+        }
+      }
+    ],
+    "PropertyRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "target": {
+        "@Core.IsURL": true,
+        "@Core.Description": "URL addressing a property [OData-URL, section 4.6] that is targeted by this recommendation"
```

Comment:
And if it is relative, it is relative to the binding parameter of the previous RecommendationsFunction call?

Reply:
Yes, see my new commit.